pkg/migration/migration_test.go

```diff
@@ -23,8 +17,240 @@ import (
+	}
+
+	expectedStcImages = ImageConfig{
+		PortworxImage:       "portworx/oci-monitor:2.9.1",
+		CustomImageRegistry: "test",
+		Components: manifest.Release{
+			Stork:     "a/stork:2.7.0",
+			Autopilot: "b/autopilot:1.3.1",
+			Telemetry: "c/ccm-service:3.0.9",
+		},
+	}
+
+	testImageMigration(t, dsImages, expectedStcImages)
+}
+
+func testImageMigration(t *testing.T, dsImages, expectedStcImages ImageConfig) {
+	clusterName := "px-cluster"
+	ds := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "portworx",
+			Namespace: "portworx",
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "portworx",
+							Image: dsImages.PortworxImage,
+							Args: []string{
+								"-c", clusterName,
+							},
+						},
+						{
+							Name:  pxutil.TelemetryContainerName,
+							Image: dsImages.Components.Telemetry,
+						},
+					},
+				},
+			},
+		},
+	}
+	storkDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      storkDeploymentName,
+			Namespace: ds.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Image: dsImages.Components.Stork,
+						},
+					},
+				},
+			},
+		},
+	}
+	autopilotDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      component.AutopilotDeploymentName,
+			Namespace: ds.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Image: dsImages.Components.Autopilot,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	k8sClient := testutil.FakeK8sClient(
+		ds, storkDeployment, autopilotDeployment,
+	)
+
+	mockController := gomock.NewController(t)
+	driver := testutil.MockDriver(mockController)
+	ctrl := &storagecluster.Controller{
+		Driver: driver,
+	}
+	ctrl.SetKubernetesClient(k8sClient)
+	mockManifest := mock.NewMockManifest(mockController)
+	manifest.SetInstance(mockManifest)
+	mockManifest.EXPECT().CanAccessRemoteManifest(gomock.Any()).AnyTimes()
+
+	driver.EXPECT().GetSelectorLabels().Return(nil).AnyTimes()
+	driver.EXPECT().String().Return("mock-driver").AnyTimes()
+
+	migrator := New(ctrl)
+
+	go migrator.Start()
+	time.Sleep(2 * time.Second)
+
+	cluster := &corev1.StorageCluster{}
+	err := testutil.Get(k8sClient, cluster, clusterName, ds.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, expectedStcImages.PortworxImage, cluster.Spec.Image)
+	require.Equal(t, expectedStcImages.CustomImageRegistry, cluster.Spec.CustomImageRegistry)
+	require.Equal(t, expectedStcImages.Components.Stork, cluster.Status.DesiredImages.Stork)
+	require.Equal(t, expectedStcImages.Components.Autopilot, cluster.Status.DesiredImages.Autopilot)
+	require.Equal(t, expectedStcImages.Components.Telemetry, cluster.Status.DesiredImages.Telemetry)
+
+	cm := &v1.ConfigMap{}
```

Comment:
We should check for non-airgapped envs as well to verify that we don't create a configmap.